### PR TITLE
Everywhere: Make JSON serialization fallible

### PR DESCRIPTION
**Pull request by IdanHo**
This allows us to eliminate a major source of infallible allocation in the Kernel, as well as lay down the groundwork for OOM fallibility in userland.

(Apologies in advance for the last commit :sweat_smile:)

### CODE REVIEW

File: AK/JsonArraySerializer.h [hunk 1]
```diff
@@ -1,139 +1,233 @@
...
+        if constexpr (IsLegacyBuilder<Builder>) {
+            TRY(m_builder.try_append('"'));
+            TRY(m_builder.try_append_escaped_for_json(value));
+            TRY(m_builder.try_append('"'));
+        } else {
+            TRY(m_builder.append('"'));
+            TRY(m_builder.append_escaped_for_json(value));
+            TRY(m_builder.append('"'));
+        }
+        return {};
```
Comment (elcuco) [inline AK/JsonArraySerializer.h:72]: This seems like using exceptions, but with another extra step. 

Just saying.

Comment (bgianfo) [reply AK/JsonArraySerializer.h:72]: Please avoid leaving useless, unactionable comments on PRs.

File: Kernel/Devices/DeviceManagement.cpp [hunk 2]
```diff
@@ -109,6 +109,15 @@ void DeviceManagement::for_each(Function<void(Device&)> callback)
     });
 }
 
+ErrorOr<void> DeviceManagement::try_for_each(Function<ErrorOr<void>(Device&)> callback)
+{
+    return m_devices.with([&](auto& map) -> ErrorOr<void> {
+        for (auto& entry : map)
+            return callback(*entry.value);
```
Comment (bgianfo) [inline Kernel/Devices/DeviceManagement.cpp:116]: Shouldn't this be just a `TRY(callback(*entry.value))`, as is the enumeration will on process the first entry?

**Review by bgianfo — APPROVED** [review]

File: Kernel/Devices/DeviceManagement.cpp (hunk 2 above)
Comment (IdanHo) [reply Kernel/Devices/DeviceManagement.cpp:116]: Absolutely, good catch, fixed

**Pull request merged**